cql3/statements/alter_keyspace_statement.cc

```diff
@@ -84,11 +84,16 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
 cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_processor& qp, const query_options&, api::timestamp_type ts) const {
     try {
-        auto old_ksm = qp.db().find_keyspace(_name).metadata();
+        auto ks = qp.db().find_keyspace(_name);
+        auto ks_md = ks.metadata();
         const auto& tm = *qp.proxy().get_token_metadata_ptr();
         const auto& feat = qp.proxy().features();
 
-        auto m = service::prepare_keyspace_update_announcement(qp.db().real_database(), _attrs->as_ks_metadata_update(old_ksm, tm, feat), ts);
+        if (ks.get_replication_strategy().uses_tablets()) {
+            return make_exception_future<std::tuple<::shared_ptr<::cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(
+                exceptions::invalid_request_exception("ALTER KEYSPACE statement is not yet supported for keyspaces with tablets."));
```

Comment:
1. Why not do it in `validate()`?  We already reject the alter_keyspace_statement for other reasons there.
2. In theory, one could still use ALTER KEYSPACE to modify `durable_writes` or even the `initial` number of tablets, so in addition to `uses_tablets()` we can also check `!_attrs->get_replication_options().empty()` for rejecting the query.

While at it, we can also check, by using `_attrs->get_map(KW_TABLETS)` if the `tablets` option is attempted to be changed (on any keyspace, not just ones that currently used tablets) and reject enabling/disabling tablets on existing keyspaces since it's not supported yet (in the future we plan to support conversion of vnode-based keyspaces to tablets, and it can be done, in theory, using ALTER KEYSPACE).

Reply:
1. I moved the `if` to the `validate()` method. Perhaps I was too biased with https://github.com/scylladb/scylladb/pull/16723, where the keyspace-tablets check is done in `prepare_schema_mutations()`.
2. I think it's simpler and less confusing to just reject everything. That's especially true when explaining the current behavior to the users in the docs.